src/async_client.cpp

```diff
@@ -480,7 +484,7 @@ itoken_ptr async_client::subscribe(const std::string& topicFilter, int qos)
 
 	response_options opts(dynamic_cast<token*>(tok.get()));
 
-	int rc = MQTTAsync_subscribe(cli_, (char*) topicFilter.c_str(), qos, &opts.opts_);
+	int rc = MQTTAsync_subscribe(cli_, static_cast<const char*>(topicFilter.c_str()), qos, &opts.opts_);
```

Comment:
Most of these `(char*)` casts were really `const_cast<char*>` since, at the time of writing this, the C lib used non-const `char*` for all of its string values. Now that the C API has changed for the better, the casts can be removed.